engine/src/flutter/lib/ui/painting.dart

```diff
@@ -5517,6 +5614,21 @@ base class FragmentShader extends Shader {
     _floats = _constructor(_program, _program._uniformFloatCount, _program._samplerCount);
   }
 
+  List<UniformFloatSlot> _getSlotsForUniform(String name, int size) {
+    final int dataSize = _program._getUniformVectorSize(name);
```

Comment:
This implementation is causing 2 lookups by name, can they be combined? It's an internal function so what about having the getIndex function take the size and it gets verified before returning the index (size of 0 or -1 to indicate "I don't care about the size")?